### DISCUSSION TITLE
Guide on porting hooks

### DISCUSSION THREAD

**Discussion by AndreasNel** [react-hookz/web#884] [Q&A] [answered]
When porting hooks from `react-use`, is the goal to maintain the same API as far as possible? Or is it up to whoever is porting it to make a decision and then it gets discussed during the PR review?

**Reason for asking:**
I was thinking of porting the [useTimeoutFn](https://github.com/streamich/react-use/blob/master/src/useTimeoutFn.ts) hook as a `useTimeoutEffect` hook, but I was wondering if we should keep the exact return values. I don't see the use of the `isReady` return variable, but I do think that the `clear` and `set` variables could _maybe_ be useful (as `cancel` and `reset`, respectively).

Maybe we could add a sentence/guide on how to go about this process as part of the `CONTRIBUTING.MD` file?

**Comment by xobotyi** [comment, marked as answer]
Everything is up to developer, we're not targeting to provide same api, as some of apis there questionable.
